public/diagnostic.js

```diff
@@ -8,6 +8,16 @@
 //  -  a quantity
 //  -  a computed property called `orderPrice`, equal to price times quantity
 //
+
+const Order = Ember.Order.extend({
```

Comment:
It looks like you're confusing Order and Cart. Make sure you're reading directions carefully and taking things one step at a time.

```js

const Order = Ember.Object.extend({
  price: this.price,
  quantity: this.quantity,
  orderPrice: Ember.computed('price', 'quantity', function(){
    return this.get('price') * this.get('quantity');
  })
});

const Cart = Ember.Object.extend({
  orders: [],
  addToCart: function(order){
    this.get('orders').addObject(order);
  },
  totalPrice: Ember.computed('orders','orders.@each.orderPrice', function(){
    let orders = this.get('orders');
    return orders.map((order) => order.get('orderPrice')).reduce((m,e) => m+e);
  })
});

let myCart = Cart.create();
myCart.addToCart(Order.create({price: 5, quantity: 2}));
myCart.addToCart(Order.create({price: 20, quantity: 1}));
myCart.addToCart(Order.create({price: 8, quantity: 3}));
console.log(myCart.get('totalPrice'));
```